mkat_fpga_tests/__init__.py

```diff
@@ -173,7 +163,7 @@ def start_correlator(self, retries=30, loglevel='INFO'):
             if array_list_messages:
                 self.array_number = array_list_messages[0].arguments[0]
                 self.rct.req.array_halt(self.array_number)
-        except IndexError:
+        except Exception:
```

Comment:
Catching Exception is probably to broad -- it may hide other errors. You need to know exactly what exception will be if the array is already cleared and only catch them.